src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java

```diff
@@ -845,9 +859,19 @@ void readEvents(final List<O> evtOps) {
 				val endpointUri = endpointCache.computeIfAbsent(nodeAddr, this::createEndpointUri);
 				val clientConfig = clientConfigCache.computeIfAbsent(endpointUri, this::createClientConfig);
 				val streamName = extractStreamName(anyEvtOp.dstPath());
+				Stream stream = new StreamImpl(scopeName, streamName);
 				val readerGroupConfigBuilder = evtReaderGroupConfigBuilder.get();
 				val readerGroupConfig = evtReaderGroupConfigCache.computeIfAbsent(
-					scopeName + SLASH + streamName, key -> readerGroupConfigBuilder.stream(key).build());
+					scopeName + SLASH + streamName, key -> {
+							StreamManager streamManager = StreamManager.create(clientConfig);
+							StreamInfo streamInfo = streamManager.getStreamInfo(scopeName, streamName);
+							StreamCut streamCut = tailReadFlag ? streamInfo.getTailStreamCut() : streamInfo.getHeadStreamCut();
+							return readerGroupConfigBuilder
+								.stream(Stream.of(scopeName, streamName),
+									streamCut,
+									StreamCut.UNBOUNDED)
```

Comment:
Am I right you're building a config for reader group which readers would read events from Stream.of(scopeName, streamName) up to streamCut?

Reply:
.stream() takes 3 parameters to configure a reader group: 1) scope/stream to read from, 2) left-border streamCut to read from, 3) right-border streamCut to read to. Here we set left border either to start of the stream or the end and we don't set any restrictions on the right border.